src/adv.py

```diff
@@ -37,15 +55,58 @@
 # Main
 #
 
-# Make a new player object that is currently in the 'outside' room.
+    # Make a new player object that is currently in the 'outside' room.
+player = Player(room['outside'])
+
+"""
+this method takes two inputs
+direction is the direction the user provides
+current room is the place player current is
+this method returns the new room if there was a room to move
+means it the move was successful or returns the current room 
+if the move wasn't successful
+"""
+def diff_directions(direction, current_room):
+    move = direction + '_to'
+    # to check if the direction is valid
+    if hasattr(current_room, move):
+        # if the direction is valid it returns the new current room
+        return getattr(current_room, move)
+    else:
+        print("There is'nt a room in that direction!")
+        return current_room
 
+     
+   
 # Write a loop that:
-#
-# * Prints the current room name
-# * Prints the current description (the textwrap module might be useful here).
-# * Waits for user input and decides what to do.
-#
-# If the user enters a cardinal direction, attempt to move to the room there.
-# Print an error message if the movement isn't allowed.
-#
-# If the user enters "q", quit the game.
+while True:
```

Comment:
Good work so far Manjula. Some of the times I tested your code and put in an invalid direction, it would return "NoneType" object has no attribute 'name'. So that makes me think that some of the error handling isn't working properly for some reason. 

Good job on getting the item object started. You might want to refactor your input command so that you are going to be able to accept multiple arguments so you can pick up items. But it looks like you're doing great so far.